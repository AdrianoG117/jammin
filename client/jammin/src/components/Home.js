import React from 'react';
import {Link} from 'react-router-dom';
// import FindJam from './FindJam';
// import CreateJam from './CreateJam';
// import {BrowserRouter as Router,Route, Switch} from 'react-router-dom';


function Home() {
  return (
    <div className="main">
        <Link to='/createjam'>
          <button className="home-btn">CREATE YOUR JAM</button>
        </Link>
        <Link to='/findjam'>
          <button className="home-btn">FIND A JAM</button>
        </Link>
      </div>
  )
}

export default Home
