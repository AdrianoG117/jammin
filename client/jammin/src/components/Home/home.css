.main {
  width: 100%;
  height: 80%;
  display: flex;
  margin-top: 3rem
}

.btn-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 50%;
  height: 80%
}

.slogan {
  width: 50%;
  height: 100%;
  position: relative
}

.slogan h1 {
  width: 40%;
  letter-spacing: 0.4rem;
  position: relative;
  left: 60px;
  font-weight: bold;
  color: white
}

.home-btn {
  width: 30vw;
  height: 18vh;
  border-radius: 15px;
  margin: 2rem;
  background-color: rgba(0, 0, 0, 0.7);
  color: white;
  font-size: 1.5rem
}

.home-btn:hover{
  background-color: rgba(0, 0, 0, 0.3)
}
