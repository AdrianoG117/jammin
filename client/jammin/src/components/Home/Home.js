import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import './home.css';

function Home(props) {
  const setJams = props.setJams;
  const setHasSearch = props.setHasSearch;

  useEffect(() => {
    setJams([]);
    setHasSearch(false);
  }, []);

  return (
    <div className="main">
      <div className="slogan">
        <h1>JAMMIN'</h1>
        <h2>Music is meant to be shared.</h2>
      </div>
      <div className="btn-container">
        <Link to="/createjam">
          <button className="home-btn">CREATE YOUR JAM</button>
        </Link>
        <Link
          to={{
            pathname: '/findjam',
          }}
        >
          <button className="home-btn">FIND A JAM</button>
        </Link>
      </div>
    </div>
  );
}

export default Home;
