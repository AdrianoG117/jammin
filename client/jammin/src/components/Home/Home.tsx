import React, { useEffect } from "react";
import { useHistory } from "react-router-dom";
import "./home.css";

function Home({setJams,setHasSearch }) {
  
  const history = useHistory();

  useEffect(() => {
      setJams([]);
      setHasSearch(false);
    
  }, []);

  return (
    <div className="main">
      <div className="slogan">
        <h1>JAMMIN'</h1>
        <h2>Music is meant to be shared.</h2>
      </div>
      <div className="btn-container">
        
          <button className="home-btn" onClick={()=> history.push("/createjam")}>CREATE YOUR JAM</button>
       
          <button className="home-btn" onClick={()=> history.push("/findjam")}>FIND A JAM</button>
        
      </div>
    </div>
  );
}

export default Home;
