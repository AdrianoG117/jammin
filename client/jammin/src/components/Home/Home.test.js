import {render, screen} from '@testing-library/react';
import Home from './Home';


describe('Home component', ()=> {
  
  test('should match the snapshot', () => {
    const {container} = render(<Home jams={[]}
                  setJams={()=>{}}
                  setHasSearch={()=>{}} />);
    expect(container.firstChild).toMatchSnapshot();
  })
  
  test('Should render the headings', () => {
    render (<Home jams={[]}
                  setJams={()=>{}}
                  setHasSearch={()=>{}} />);
    //title
    screen.getByText(/JAMMIN'/)
    screen.getByText(/Music is meant to be shared./)
    screen.getByText(/CREATE YOUR JAM/)
    screen.getByText(/FIND A JAM/)
  })

})