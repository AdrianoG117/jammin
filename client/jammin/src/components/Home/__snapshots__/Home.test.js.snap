// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match the snapshot 1`] = `
<div
  class="main"
>
  <div
    class="slogan"
  >
    <h1>
      JAMMIN'
    </h1>
    <h2>
      Music is meant to be shared.
    </h2>
  </div>
  <div
    class="btn-container"
  >
    <button
      class="home-btn"
    >
      CREATE YOUR JAM
    </button>
    <button
      class="home-btn"
    >
      FIND A JAM
    </button>
  </div>
</div>
`;
